yt_dlp/extractor/asobistage.py

```diff
@@ -0,0 +1,160 @@
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        },
+    }, {
+        'url': 'https://asobistage.asobistore.jp/event/ijigenfes_utagassen/player/day1',
+        'only_matching': True,
+    }]
+
+    def _check_login(self, video_id):
+        check_login_json = self._download_json(
+            'https://asobistage-api.asobistore.jp/api/v1/check_login', video_id, expected_status=(400),
+            note='Checking login status', errnote='Unable to check login status')
+        error = traverse_obj(check_login_json, ('payload', 'error_message'), ('error'), get_all=False)
+
+        if not error:
+            return
+        elif error == 'notlogin':
+            self.raise_login_required(metadata_available=False)
+        else:
+            raise ExtractorError(f'Unknown error: {error}')
+
+    def _get_token(self, video_id):
+        return self._search_regex(r'\"([^"]+)\"', self._download_webpage(
+            'https://asobistage-api.asobistore.jp/api/v1/vspf/token', video_id,
+            note='Getting token', errnote='Unable to get token'),
+            name="token")
+
+    def _get_owned_tickets(self, video_id):
+        for url, name in [
+            ('https://asobistage-api.asobistore.jp/api/v1/purchase_history/list', 'ticket purchase history'),
+            ('https://asobistage-api.asobistore.jp/api/v1/serialcode/list', 'redemption history'),
+        ]:
+            for id in traverse_obj(self._download_json(
+                    url, video_id, note=f'Downloading {name}', errnote=f'Unable to download {name}'),
+                    ('payload', 'value', ..., 'digital_product_id', {str})):
+                yield id
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        video_type_name = self._match_valid_url(url).group('type')
+        webpage = self._download_webpage(url, video_id)
+
+        self._check_login(video_id)
+
+        video_type_id, live_status = {
+            'archive': ['archives', 'was_live'],
+            'player': ['broadcasts', 'is_live'],
+        }.get(video_type_name) or [None, None]
+        if not video_type_id:
+            raise ExtractorError('Unknown video type')
+
+        event_data = self._search_nextjs_data(webpage, video_id)
+        event_id = traverse_obj(event_data, ('query', 'event', {str}))
+        event_slug = traverse_obj(event_data, ('query', 'player_slug', {str}))
+        event_title = traverse_obj(event_data, ('props', 'pageProps', 'eventCMSData', 'event_name', {str}))
+        if not all((event_id, event_slug, event_title)):
+            raise ExtractorError('Unable to get required event data')
+
+        channel_list_url = functools.reduce(urljoin, [
+            self._search_regex(
+                r'"(?P<url>https://asobistage\.asobistore\.jp/cdn/[^/]+/)', webpage,
+                'cdn endpoint url', group=('url')),
+            'events/', f'{event_id}/', f'{video_type_id}.json'])
```

Comment:
As far as I can tell, `https://asobistage.asobistore.jp/cdn/v101/events/` has been hardcoded in the JS since at least 2022.
I would just write `f'https://asobistage.asobistore.jp/cdn/v101/events/{event_id}/{video_type_id}.json'`. Any reason not to?